### PR TITLE
Fewer bindles, better registries

### PR DESCRIPTION
**Pull request by itowlson**
This PR updates the Spin "Packaging and Distribution" documentation to refer to registries.

* The section about Bindle is entirely removed.
  * I also removed references from the Contribution page, but have _not_ tackled the URL Shortener tutorial.
* References to registry support as 'new', or to the version in which it was introduced, are removed.
* A lot of talk about OCI is rephrased, placed behind a "what's all this about containers" expander, or both.
* References to `spin registry run` are updated to `spin up -f`.
* The content on the registry commands is rephrased and refocused.
  * I have _not_ tackled the "tutorial" - @macolso we need to figure out the relationship between these two pages.
* The page is moved out of the Advanced section of the site TOC into a proposed new Applications section.
  * If this is controversial I am happy to back it out of this PR; we can think about how to organise the TOC separately.

Screenshot of the TOC change:

image

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)

Screenshot 2023-03-15 at 2 19 59 pm

- [x] Has run `npm run build-index`
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)

Screenshot 2023-03-15 at 2 18 16 pm

Screenshot 2023-03-15 at 2 17 13 pm

- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors

Screenshot 2023-03-15 at 2 13 36 pm

Screenshot 2023-03-15 at 2 16 05 pm

### CODE REVIEW

File: content/spin/distributing-apps.md [hunk 1]
```diff
@@ -1,104 +1,123 @@
...
+Verification for ghcr.io/alyssa-p-hacker/hello-world@sha256:06b19 --
+The following checks were performed on each of these signatures:
+  - The cosign claims were validated
+  - Existence of the claims in the transparency log was verified offline
+  - Any certificates were verified against the Fulcio roots.
 
-The application can also be prepared in a local directory before pushing to the registry by running `spin bindle prepare`.
\ No newline at end of file
+# The consumer of your app can now run it from the registry.
+$ spin up -f ghcr.io/alyssa-p-hacker/hello-world:v1
```
Comment (itowlson) [inline content/spin/distributing-apps.md:122]: @radu-matei are you happy for this to go out - I have reservations about implying the mutable tag is now verified but maybe `cosign` is doing something cunning to make it safe?

Comment (itowlson) [reply content/spin/distributing-apps.md:122]: Had a word with Radu and he wants this to go out as is, until we have pull by digest at which point we can update it.

File: content/spin/distributing-apps.md [hunk 2]
```diff
@@ -1,104 +1,133 @@
...
-# You can now point Spin to the application in the registry and run it.
-$ spin registry run ghcr.io/radu-matei/spin-hello-world:v1
+<!-- @nocpy -->
+
+```bash
+$ spin registry push ghcr.io/alyssa-p-hacker/hello-world:v1
 ```
 
-## Distributing Spin Applications Using Bindle
+Notice that the user name is part of the reference; the registry does not infer it from the login.  Also notice that the version is specified explicitly; Spin does not infer it from the `spin.toml` file.
```
Comment (vdice) [inline content/spin/distributing-apps.md:66]: nit: consistency -- "user name" vs "username" as used elsewhere..

File: content/spin/distributing-apps.md [hunk 3]
```diff
@@ -1,104 +1,133 @@
...
+<!-- @nocpy -->
+
+```bash
+# Push your Spin application to any registry that supports the OCI registry artifacts,
+# such as the GitHub Container Registry, Docker Hub, Azure ACR, or AWS ECR.
+$ spin registry push ghcr.io/alyssa-p-hacker/hello-world:v1
+
+# You can now sign your Spin app using Cosign (or any other tool that can sign
+# OCI registry objects).
+$ COSIGN_EXPERIMENTAL=1 cosign sign ghcr.io/alyssa-p-hacker/hello-world@sha256:06b19
```
Comment (vdice) [inline content/spin/distributing-apps.md:116]: nit: I believe we can drop use of this env var if we note that [cosign v2.0.0+](https://blog.sigstore.dev/cosign-2-0-released/) is used.

File: content/spin/distributing-apps.md [hunk 4]
```diff
@@ -1,104 +1,133 @@
...
+> Downloaded applications are cached. When run, or pulled again, Spin checks to see if they have changed from the cached copy, and downloads only the changes if any.
+
+## Signing Spin Applications and Verifying Signatures
+
+Because Spin uses the container registry standards to distribute applications, it can also take advantage of tooling built around those standards.  Here is an example of using [Cosign and Sigstore](https://docs.sigstore.dev/cosign/overview/) to sign and verify a Spin application:
+
+<!-- @nocpy -->
+
+```bash
+# Push your Spin application to any registry that supports the OCI registry artifacts,
```
Comment (vdice) [inline content/spin/distributing-apps.md:110]: nano-nit: Elsewhere we are in the habit of supplying informational/instructional test in markdown, interspersed with bash snippets.  We might consider updating this section to follow suit.

File: content/spin/distributing-apps.md [hunk 5]
```diff
@@ -1,104 +1,133 @@
...
-<!-- @nocpy -->
+### Logging In Using a Token
+
+In a non-interactive environment such as GitHub Actions, you will typically log in using a token configured in the environment settings, rather than a password.  To do this, use the `--password-stdin` flag, and `echo` the token value to the login command's standard input.  This example shows logging into GHCR from a GitHub action:
+
+<!-- @noCpy -->
 
 ```bash
-$ spin registry run ghcr.io/radu-matei/spin-hello-world:v1
+$ echo "${{ secrets.GITHUB_TOKEN }}" | spin registry login ghcr.io --username ${{ github.actor }} --password-stdin
```
Comment (vdice) [inline content/spin/distributing-apps.md:43]: I think some extra escaping for the `${{...}}` values may be necessary here.  I'm currently seeing this section render as follows:
Screenshot 2023-03-14 at 3 06 29 PM

**Review by vdice — COMMENTED** [review]
Excellent docs additions!

File: content/spin/distributing-apps.md (hunk 5 above)
Comment (itowlson) [reply content/spin/distributing-apps.md:43]: Yikes, completely missed that.  Thanks!

File: content/spin/distributing-apps.md (hunk 4 above)
Comment (itowlson) [reply content/spin/distributing-apps.md:110]: I am going to punt on this for now if that is okay.

**Review by vdice — APPROVED** [review]
👏

**Review by tpmccallum — APPROVED** [review]
LGTM!

**Pull request merged**